pkg/machines/components/nicBody.jsx

```diff
@@ -0,0 +1,186 @@
+            availableSources = networks.map(network => network.name);
+        else if (dialogValues.networkType === "direct")
+            availableSources = networkDevices;
+
+        if (availableSources.length > 0) {
+            networkSourcesContent = availableSources
+                    .map(networkSource => {
+                        return (
+                            <Select.SelectEntry data={networkSource} key={networkSource}>
+                                {networkSource}
+                            </Select.SelectEntry>
+                        );
+                    });
+        } else {
+            if (dialogValues.networkType === "network")
+                defaultNetworkSource = _("No Virtual Networks");
+            else if (dialogValues.networkType === "direct")
+                defaultNetworkSource = _("No Network Devices");
+
+            networkSourcesContent = (
+                <Select.SelectEntry data='empty-list' key='empty-list'>
+                    {defaultNetworkSource}
+                </Select.SelectEntry>
+            );
+            networkSourceEnabled = false;
+        }
+    }
+
+    return (
+        <>
+            <label className='control-label' htmlFor={`${idPrefix}-select-type`}>
+                {_("Interface Type")}
+            </label>
+            <Select.Select id={`${idPrefix}-select-type`}
+                           onChange={value => onValueChanged('networkType', value)}
+                           initial={defaultNetworkType}
+                           extraClass='form-control'>
+                {availableNetworkTypes
+                        .map(networkType => {
+                            return (
+                                <Select.SelectEntry data={networkType.name} key={networkType.name} disabled={networkType.disabled || false}>
+                                    {networkType.desc}
+                                </Select.SelectEntry>
+                            );
+                        })}
+            </Select.Select>
+            {["network", "direct"].includes(dialogValues.networkType) && (
+                <div className='ct-form'>
+                    <label className='control-label' htmlFor={`${idPrefix}-select-source`}>
+                        {_("Source")}
+                    </label>
+                    <Select.Select id={`${idPrefix}-select-source`}
+                                   onChange={value => onValueChanged('networkSource', value)}
+                                   enabled={networkSourceEnabled}
+                                   initial={defaultNetworkSource}
+                                   extraClass='form-control'>
+                        {networkSourcesContent}
+                    </Select.Select>
+                </div>
+            )}
+        </>
+    );
+};
+
+export const NICBody = ({ idPrefix, dialogValues, onValueChanged, networks, interfaces, nodeDevices, connectionName, osTypeArch, osTypeMachine }) => {
```

Comment:
I don't see why this component should be independent.
Just export the NetworkTypeAndSourceRow and the NetworkModelRow and use them from the Creation for NIC component.